torch/distributed/distributed_c10d.py

```diff
@@ -588,22 +597,36 @@ def _new_process_group_helper(world_size,
         prefix_store = PrefixStore(group_name, store)
 
         if backend == Backend.GLOO:
+            if pg_options is None:
+                pg_options = ProcessGroupGloo.Options()
+                pg_options.timeout = timeout
+                ifname_env = getenv("GLOO_SOCKET_IFNAME")
+                if ifname_env is not None:
+                    pg_options.devices = [ProcessGroupGloo.create_device(interface=iface) for iface in ifname_env.split(",")]
```

Comment:
Does this mean if user creates their own `ProcessGroupGloo.Options()`, they are responsible for create device from interface name themselves? 

It looks like currently for gloo, we only provide `timeout` as a customizable option and always create devices in C++ similar to what we're doing here.

Should we remove `devices` arg from public interface of ProcessGroupGlooOptions and manually populate it always ourselves like we do currently? We can revisit allowing users to set it if there is a requirement.